source/core/Events.cpp

```diff
@@ -414,13 +423,15 @@ class DynamicEventRegInfo {
     // If the registered data is a cluster or nested cluster updated *dynIndexBase with the nested element
     // index as it traverses, and returns a non-zero index if a match occcurs.  (The dynIndex value is
     // the same one storied in the EventSpec in the event structure configuration data.)
-    Int32 DynamicEventMatchCore(TypeRef regRefType, void *pData, RefNum refnum, Int32 *dynIndexBase) {
+    static Int32 DynamicEventMatchCore(TypeRef regRefType, void *pData, RefNum refnum, Int32 *dynIndexBase) {
         Int32 dynIndex = 0;
         if (regRefType->IsRefnum() && pData && (static_cast<RefNumVal*>(pData)->GetRefNum() == refnum)) {
             dynIndex = *dynIndexBase;
             return dynIndex;
-        } else if (regRefType->IsArray() && regRefType->Rank() == 1 && regRefType->GetSubElement(0)->IsRefnum()) {
-            TypedArray1D<RefNumVal> *aRef = *(TypedArray1D<RefNumVal>**)pData;
+        }
+
```

Comment:
I don't agree with the 'extraneous else' rule.
It doesn't hurt anything and helps readability, because you don't have to look inside the previous clauses for early returns to understand control-flow.

Reply:
You still need to look inside the if statement to see if a return statement is being used because there may be code after the if-elseif chain, and that code will not be executed if the first if (with the early return) gets executed. That's not the case in this function, but generally it is possible.

Reply:
Regardless, the else if helps readability.  Leave it.